File: 30 Days of Code/30 - Day 29 - Bitwise AND.py
# ========================
#       Information
# ========================

# Direct Link: https://www.hackerrank.com/challenges/30-bitwise-and/problem
# Difficulty: Medium
# Max Score: 30
# Language: Python

# ========================
#         Solution
# ========================

if __name__ == '__main__':
    T = int(input())

    for t_itr in range(T):
        nk = input().split()

        n = int(nk[0])

        k = int(nk[1])

        count = 0

        for x in range(n, 1, -1):
            for y in range(x-1, 0, -1):
                kk = x & y
                if kk > count and kk < k:
                    count = kk
                if count == k - 1:
                    break
            if count == k - 1:
                break

        print(count)

File: 30 Days of Code/C#/30 - Day 29 - Bitwise AND.cs
// ========================
//       Information
// ========================

// Direct Link: https://www.hackerrank.com/challenges/30-bitwise-and/problem
// Difficulty: Medium
// Max Score: 30
// Language: C#

// ========================
//         Solution
// ========================

using System;

class Solution {
    static void Main(String[] args) {
        int t = Convert.ToInt32(Console.ReadLine());

        for (int i = 0; i < t; i++) {
            string[] tokens_n = Console.ReadLine().Split(' ');
            int n = Convert.ToInt32(tokens_n[0]);
            int k = Convert.ToInt32(tokens_n[1]);

            if (((k - 1) | k) > n && k % 2 == 0) {
                Console.WriteLine(k - 2);
            }
            else {
                Console.WriteLine(k - 1);
            }
        }
    }
}
